finder.py

```diff
@@ -0,0 +1,6 @@
+@app.route('/search', methods=['POST'])
+def search():
+
+    name = request.form['name']
+    animals = Animal.query.filter_by(name=name).all()

```

Comment:
To make possibility to search via substring you can use
.query.filter(Animal.name.ilike(name))